include/mega/heartbeats.h

```diff
@@ -118,12 +88,71 @@ class HeartBeatBackupInfo : public CommandListener
     void updateLastActionTime();
 };
 
+/**
+ * @brief Backup info controlled by transfers
+ * progress is advanced with transfer progress
+ * by holding the count of total and transferred bytes
+ *
+ */
+class HeartBeatTransferProgressedInfo : public HeartBeatBackupInfo
+{
+public:
+    HeartBeatTransferProgressedInfo(handle backupId);
+    MEGA_DISABLE_COPY(HeartBeatTransferProgressedInfo)
+
+    virtual void updateTransferInfo(MegaTransfer *transfer);
+    virtual void removePendingTransfer(MegaTransfer *transfer);
+    virtual void clearFinshedTransfers();
+    virtual void setTotalBytes(long long value);
+    virtual void setTransferredBytes(long long value);
+
+    virtual double progress() const override;
```

Comment:
```suggestion
    double progress() const override;
```

Reply:
potential child class may override this one :thinking:

Reply:
Removed after some philosophical discussion. Agreed that `virtual` is unnecessary and the advantage of having it to prevent some very rare situation is outweighed by the unlikeliness of that.